src/Core/Managed/Shared/TelemetryClientExtensions.cs

```diff
@@ -93,26 +94,78 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
                 operationTelemetry.GenerateOperationId();
             }
 
+            var telemetryContext = operationTelemetry.Context.Operation;
+
             // If the operation is not executing in the context of any other operation
             // set its name and id as a context (root) operation name and id
-            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Id))
+            if (string.IsNullOrEmpty(telemetryContext.Id))
             {
-                operationTelemetry.Context.Operation.Id = operationTelemetry.Id;
+                telemetryContext.Id = operationTelemetry.Id;
             }
 
-            if (string.IsNullOrEmpty(operationTelemetry.Context.Operation.Name))
+            if (string.IsNullOrEmpty(telemetryContext.Name))
             {
-                operationTelemetry.Context.Operation.Name = operationTelemetry.Name;
+                telemetryContext.Name = operationTelemetry.Name;
             }
 
+            bool isActivityEnabled = false;
+#if !NET40
+            if (ActivityExtensions.IsActivityEnabled())
+            {
+                bool operationNameIsSet = false;
+
+                var childActivity = new Activity("Microsoft.ApplicationInsights.OperationContext");
```

Comment:
Operation Name in ApplicationInsights is quite unfortunately not the same as Activity.OperationName,

The reason is that AppInsights for ASP.NET classic uses incoming request routes for the Operation.Name. And the routes are calculated somewhere late in the pipeline, after Activity for the request is started.
Activity Name could only be provided in ctor and could never change. 
Also Activity does not support null or empty names.

So AI will use special Tag  "OperationName" for it's own  operation name.